google/resource_storage_bucket.go

```diff
@@ -77,6 +79,65 @@ func resourceStorageBucket() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"lifecycle_rule": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"action": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MaxItems: 1,
+							Set:      resourceGCSBucketLifecycleRuleActionHash,
```

Comment:
I had to dig a bit to find out the constraints. The XML API doc looks a bit more precise the JSON API about those facts, but I may be wrongly mixing those two.

From the XML API :

> Action	Defines the action to be taken, which must contain one and only one specific action element. Required.

cf. https://cloud.google.com/storage/docs/xml-api/put-bucket-lifecycle#request_body_elements

From the JSON API:

> A lifecycle management rule, which is made of an action to take and the condition(s) under which the action will be taken.